tools/osv-linter/internal/checks/packages.go

```diff
@@ -0,0 +1,173 @@
+package checks
+
+import (
+	"fmt"
+	"slices"
+	"strings"
+
+	"github.com/ossf/osv-schema/linter/internal/helpers"
+	"github.com/package-url/packageurl-go"
+	"github.com/tidwall/gjson"
+)
+
+var CheckPackageExists = &CheckDef{
+	Code:        "P0001",
+	Name:        "package-exists",
+	Description: "package exists in ecosystem's registry",
+	Check:       PackageExists,
+}
+
+// PackageExists checks the package exists in the registry for that ecosystem.
+func PackageExists(json *gjson.Result) (findings []CheckError) {
+	affectedEntries := json.Get(`affected`)
+
+	knownExistent := make(map[string][]string)
```

Comment:
We can just use a struct as a map key here to simplify things. I.e.

```
type Package struct{
  Ecosystem string
  Name string
}

knownExistent := make(map[Package]bool) 
```

Reply:
Thanks. That much better approach never comes naturally to me :-(
Refactored in https://github.com/ossf/osv-schema/pull/243/commits/2302267a6a90d4c74234a44a7cd46f562af4b8b7